src/mesh/boundary_nonuniform.cxx

```diff
@@ -0,0 +1,1285 @@
+#include "boundary_nonuniform.hxx"
+
+#include <bout/constants.hxx>
+
+#include <boundary_standard.hxx>
+#include <boutexception.hxx>
+#include <derivs.hxx>
+#include <fft.hxx>
+#include <globals.hxx>
+#include <invert_laplace.hxx>
+#include <msg_stack.hxx>
+#include <output.hxx>
+#include <utils.hxx>
+
+void BoundaryDirichletNonUniform_O2::apply(Field3D& f, BoutReal t) {
+  bndry->first();
+
+  // Decide which generator to use
+  std::shared_ptr<FieldGenerator> fg = gen;
+  if (!fg)
+    fg = f.getBndryGenerator(bndry->location);
+
+  BoutReal val = 0.0;
+
+  Mesh* mesh = f.getMesh();
+  CELL_LOC loc = f.getLocation();
+
+  BoutReal vals[mesh->LocalNz];
+  int bx = bndry->bx;
+  int by = bndry->by;
+  int stag = 0;
```

Comment:
Is this the signed-direction of the stagger? Does it ever make sense to take on other values? It might make sense to make this be an `enum class`

Maybe also `stag` -> `stagger`